### PR TITLE
Update Nanopore Parsing

### PR DESCRIPTION
**Pull request by wow-such-code**
* revert faulty merge

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 1]
```diff
@@ -17,8 +16,23 @@ import java.util.regex.Pattern
...
+        BASE_CALLER,
+        BASE_CALLER_VERSION,
+        DEVICE_TYPE,
+        FLOWCELL_ID,
+        FLOWCELL_POSITION,
+        FLOWCELL_TYPE,
+        LIBRARY_PREPARATION_KIT,
+        MACHINE_HOST,
+        START_DATE
+    }
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:33]: We do not use that anymore

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 2]
```diff
@@ -17,8 +16,23 @@ import java.util.regex.Pattern
 @Log4j2
 final class OxfordNanoporeMeasurement {
 
+    private static final String LIBRARY_PREP_KIT_SCHEMA = "SQK-.*"
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:19]: This is contained in the metadata

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 3]
```diff
@@ -17,8 +16,23 @@ import java.util.regex.Pattern
...
+        FLOWCELL_ID,
+        FLOWCELL_POSITION,
+        FLOWCELL_TYPE,
+        LIBRARY_PREPARATION_KIT,
+        MACHINE_HOST,
+        START_DATE
+    }
 
-    private final Metadata metadata
+    private final Map<METADATA_FIELD, String> metadata
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:35]: We do not use that anymore

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 4]
```diff
@@ -31,11 +45,12 @@ final class OxfordNanoporeMeasurement {
...
         this.logFilesCollection = new ArrayList<>()
         this.folders = new HashMap<>()
+        this.metadata = new HashMap()
 
         this.measurementFolder = MeasurementFolder.create(name, path, children)
 
-        this.metadata = Metadata.from(metadata)
-
+        validateMetaData(metadata)
+        readMetaData(metadata)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:53]: this is done in the metadata

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 5]
```diff
@@ -49,6 +64,27 @@ final class OxfordNanoporeMeasurement {
...
+                "hostname",
+                "started"
+        ]
+        def missingKeys = expectedKeys.stream()
+                .filter({ !metadata.keySet().contains(it) })
+                .collect()
+        if (!missingKeys.isEmpty()) {
+            throw new IllegalArgumentException('Required metadata properties missing: ' + missingKeys.join(", "))
+        }
+    }
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:86]: we do not do this here anymore and use the schema instead

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 6]
```diff
@@ -67,6 +103,35 @@ final class OxfordNanoporeMeasurement {
...
+        this.metadata[METADATA_FIELD.BASE_CALLER] = metadata["base_caller"]
+        this.metadata[METADATA_FIELD.BASE_CALLER_VERSION] = metadata["base_caller_version"]
+        this.metadata[METADATA_FIELD.DEVICE_TYPE] = metadata["device_type"]
+        this.metadata[METADATA_FIELD.FLOWCELL_ID] = metadata["flow_cell_id"]
+        this.metadata[METADATA_FIELD.FLOWCELL_POSITION] = metadata["flow_cell_position"]
+        this.metadata[METADATA_FIELD.FLOWCELL_TYPE] = metadata["flow_cell_product_code"]
+        this.metadata[METADATA_FIELD.LIBRARY_PREPARATION_KIT] = extractLibraryKit(metadata["protocol"] ?: "")
+        this.metadata[METADATA_FIELD.MACHINE_HOST] = metadata["hostname"]
+        this.metadata[METADATA_FIELD.START_DATE] = metadata["started"]
+    }
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:118]: was moved to the metadata class

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 7]
```diff
@@ -67,6 +103,35 @@ final class OxfordNanoporeMeasurement {
...
+        Pattern pattern = Pattern.compile(LIBRARY_PREP_KIT_SCHEMA, Pattern.CASE_INSENSITIVE)
+        Matcher m = pattern.matcher(text)
+        while (m.find()) {
+            result.add(m.group())
+        }
+        if (result.isEmpty()) {
+            throw new MissingPropertyException("Could not find information about the library preparation kit.")
+        }
+        return result[0]
+    }
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:133]: was moved to the metadata class

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 8]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
      * @return
      */
     String getAdapter() {
-        return metadata.getAdapter()
+        return metadata.get(METADATA_FIELD.ADAPTER) ?: ""
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:226]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 9]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the asic temperature.
      * @return
      */
     String getAsicTemp() {
-        return metadata.getAsicTemp()
+        return metadata.get(METADATA_FIELD.ASIC_TEMP)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:234]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 10]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the device type.
      * @return
      */
     String getDeviceType() {
-        return metadata.getDeviceType()
+        return metadata.get(METADATA_FIELD.DEVICE_TYPE)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:242]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 11]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the flow cell id.
      * @return
      */
     String getFlowcellId() {
-        return metadata.getFlowcellId()
+        return metadata.get(METADATA_FIELD.FLOWCELL_ID)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:250]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 12]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the flow cell position.
      * @return
      */
     String getFlowCellPosition() {
-        return metadata.getFlowcellPosition()
+        return metadata.get(METADATA_FIELD.FLOWCELL_POSITION)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:258]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 13]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the flow cell type.
      * @return
      */
     String getFlowCellType() {
-        return metadata.getFlowcellType()
+        return metadata.get(METADATA_FIELD.FLOWCELL_TYPE)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:266]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 14]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the base caller.
      * @return
      */
     String getBaseCaller() {
-        return metadata.getBaseCaller()
+        return metadata.get(METADATA_FIELD.BASE_CALLER)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:274]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 15]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the base caller version.
      * @return
      */
     String getBaseCallerVersion() {
-        return metadata.getBaseCallerVersion()
+        return metadata.get(METADATA_FIELD.BASE_CALLER_VERSION)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:282]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 16]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the library kit used.
      * @return
      */
     String getLibraryPreparationKit() {
-        metadata.getLibraryPreparationKit()
+        return metadata.get(METADATA_FIELD.LIBRARY_PREPARATION_KIT)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:290]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 17]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the machine host name.
      * @return
      */
     String getMachineHost() {
-        metadata.getMachineHost()
+        return metadata.get(METADATA_FIELD.MACHINE_HOST)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:298]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 18]
```diff
@@ -158,87 +223,87 @@ final class OxfordNanoporeMeasurement {
...
     }
 
     /**
      * Provides access to the experiment start date.
      * @return
      */
     String getStartDate() {
-        metadata.getStartDate()
+        return metadata.get(METADATA_FIELD.START_DATE)
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:306]: please revert

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy [hunk 19]
```diff
@@ -311,123 +376,4 @@ final class OxfordNanoporeMeasurement {
     String getRelativePath() {
         return this.measurementFolder.relativePath
     }
-
-    private static class Metadata {
-        private static final Map<String, ?> SCHEMA = parseMetadataSchema()
-        private static final String LIBRARY_PREP_KIT_SCHEMA = "SQK-.*(?=:)"
```
Comment (KochTobi) [inline src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:317]: why did you remove the encapsulation?

**Review by KochTobi — CHANGES_REQUESTED** [review]

File: src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy (hunk 19 above)
Comment (wow-such-code) [reply src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy:317]: github did

**Review by KochTobi — DISMISSED** [review]

**Comment by KochTobi** [comment]
LGTM

**Review by KochTobi — APPROVED** [review]
LGTM
